languages/elixir/exercises/concept/structs/test/remote_control_car_test.exs

```diff
@@ -0,0 +1,47 @@
+defmodule RemoteControlCarTest do
+  use ExUnit.Case
+
+  test "new" do
+    car = RemoteControlCar.new()
+    assert car.battery_percentage == 100
+    assert car.distance_driven_in_meters == 0
+    assert Map.equal?(car, %RemoteControlCar{})
+  end
+
+  test "drive with battery" do
+    car = RemoteControlCar.new()
+    assert car
+      |> RemoteControlCar.drive()
+      |> Map.equal?(%RemoteControlCar{battery_percentage: 99, distance_driven_in_meters: 20})
+  end
+
+  test "drive with dead battery" do
+    car = RemoteControlCar.new()
+    car = %{car | battery_percentage: 0}
+    assert car
+      |> RemoteControlCar.drive()
+      |> Map.equal?(%RemoteControlCar{battery_percentage: 0, distance_driven_in_meters: 0})
+  end
+
+  test "display distance of new" do
+    car = RemoteControlCar.new()
+    assert RemoteControlCar.display_distance(car) == "0 meters"
+  end
+
+  test "display distance of driven" do
+    car = RemoteControlCar.new()
+    car = %{car | distance_driven_in_meters: 20}
+    assert RemoteControlCar.display_distance(car) == "20 meters"
+  end
+
+  test "display battery of new" do
+    car = RemoteControlCar.new()
+    assert RemoteControlCar.display_battery(car) == "Battery at 100%"
+  end
+
+  test "display battery of driven" do
```

Comment:
For each test of a function that is supposed to accept a `RemoteControlCar` as an argument, I would write a test that checks that maps don't work. To make sure people learn how to match a specific struct.